### PR TITLE
Yield component token tests

### PR DESCRIPTION
**Pull request by fragosti**

### CODE REVIEW

File: contracts/YieldComponentToken.sol [hunk 1]
```diff
@@ -154,24 +155,59 @@ contract YieldComponentToken is ERC20Base, Ownable {
...
+  /// @param owner Owner and future recipient of the accrued yield
+  /// @param currPrice The price of fullToken to use for the calculation. Must be more than internally stored lastPrice
+  /// @return The payout amount denoted in the decimal precision of the fullToken
+  function calculatePayoutAmount(address owner, uint256 currPrice) public view returns (uint256) {
+    uint256 lastPrice = lastPrices[owner];
+    uint256 balance = balances[owner];
+    if (balance == 0 || lastPrice == 0) {
+      return 0;
+    }
+    uint256 payoutAmount = calculatePayoutAmount(balance, currPrice, lastPrice, ERC20(fullToken).decimals());
```
Comment (fabioberger) [inline contracts/YieldComponentToken.sol:187]: Not all ERC20's are properly implemented and some might lack the `decimals()` method. I'm assuming in those cases this method would revert. If so, we should probably check for this method in the `split()` method and disallow the splitting of fullTokens that don't have it.

File: test/yield_component_token.ts [hunk 2]
```diff
@@ -204,18 +204,108 @@ describe("YieldComponentToken", () => {
       expect(await yieldComponentToken.balanceOf(address)).to.eq("246913578000000000000");
     });
   });
-  describe("withdrawYield", async () => {
-    it("should withdraw the accrued yield to msg.sender and update lastPrice", async () => {
-      // TODO(fabio): Write me!
+  describe.only("transfer", async () => {
```
Comment (fabioberger) [inline test/yield_component_token.ts:207]: Accidental `only`

File: test/yield_component_token.ts [hunk 3]
```diff
@@ -204,18 +204,108 @@ describe("YieldComponentToken", () => {
...
+    it("should allow a transfer when an allowance is set and is great enough", async () => {
+      const [ownerSigner, senderSigner] = await ethers.getSigners();
+      const [owner, sender] = await Promise.all([ownerSigner.getAddress(), senderSigner.getAddress()]);
+      const amountToSend = "12340000";
+      const { yieldComponentToken } = await deployAll("X Token", "XXX", deployedAddresses, splitVault);
+      await yieldComponentToken.mint(owner, "10000000000000");
+      await yieldComponentToken.approve(sender, amountToSend);
+      expect(await yieldComponentToken.balanceOf(sender)).to.eq(0);
+      await yieldComponentToken.connect(senderSigner).transferFrom(owner, sender, amountToSend);
+      expect(await yieldComponentToken.balanceOf(sender)).to.eq(amountToSend);
```
Comment (fabioberger) [inline test/yield_component_token.ts:285]: Nit: It's awkward that this test involves sending tokens to the `sender` address. Maybe introduce a third EOA, maybe named `receiver`?

File: contracts/YieldComponentToken.sol (hunk 1 above)
Comment (fragosti) [reply contracts/YieldComponentToken.sol:187]: Adding to constructor for both capital and yield.

File: test/yield_component_token.ts (hunk 2 above)
Comment (fragosti) [reply test/yield_component_token.ts:207]: Very WIP D:

**Pull request merged**